### PR TITLE
Susan

### PR DESCRIPTION
**Pull request by ZhouSusan**
Implemented a Chat Server with Express and Socket.io

-installed npm && socket.io dependences
-added cors and cors middleware in app.js

-created chat.js: 
constructor of Connection class:  takes in Socket.io server instance and new socket, sets up callbacks on events coming from the socket.io
 -messages object is a set that holds all messages 

-sendMessage()- send a message to all sockets from Socket.io server 

-handleMessage(): when a new message is sent, it creates a message object and adds it to messages set. Then sendMessage() is called. The setTimeout() us a callback that will delete the messaged after a fixed time.

### CODE REVIEW

File: chat-server/app.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+var cors = require('cors');
+var express = require('express');
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+
+var indexRouter = require('./routes/index');
+var usersRouter = require('./routes/users');
+
```
Comment (ProwlingLynx) [inline chat-server/app.js:9]: The docs on express router suggest you can use the router object once you've added the specific routes to it. 
http://expressjs.com/en/5x/api.html#router

Note: the router object you require in './routers/*'.js' and the router object you use in app.js are the same instance.

File: chat-server/routes/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+var express = require('express');
+var router = express.Router();
+
+/* GET home page. */
+router.get('/', function(req, res, next) {
+  res.render('index', { title: 'Express' });
```
Comment (ProwlingLynx) [inline chat-server/routes/index.js:6]: I did not see any render reference in the ./public/index.html

**Review by ProwlingLynx — COMMENTED** [review]
Aside from the comments its looking good.

File: chat-server/chat.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        this.io = io;
+
+        socket.on('getMessages', () => this.getMessages());
+        socket.on('message', (value) =>  this.handleMessage(value));
+        socket.on('disconnect', () => this.disconnet());
+        socket.on('connect_error', (err) => {
+            console.log('connect_error due to ' + err.message);
+        });
+    }
+
```
[the remark this answers is not in the document]
Comment (ZhouSusan) [reply chat-server/chat.js:18]: Yes, that is what I mean line 11 to 13 are places holders.

File: chat-server/app.js (hunk 1 above)
Comment (ZhouSusan) [reply chat-server/app.js:9]: Thank you for sharing this document. I am just trying a new approach. I will look into this though.

File: chat-server/routes/index.js (hunk 2 above)
Comment (ZhouSusan) [reply chat-server/routes/index.js:6]: The only render reference within in the ./public/index.html is to GET the home page which display the index.html.

**Pull request merged**